### ISSUE TITLE
Identifier sub-properties proposal

### ISSUE THREAD

**Issue by WeaverStever** [schemaorg/schemaorg#2778]
Similar to the ISBN (books) and ISSN (periodicals) the music industry has some developed ontologies not listed. 
https://schema.org/identifier

**Example of declared sub-property isbn**
~~~~
{
 "@context": "https://schema.org/",
 "@type": "Book",
 "name": "Library linked data in the cloud : OCLC's experiments with new models of resource description",
 "author": "Carol Jean Godby",
 "isbn": "9781627052191"
}
~~~~

I propose adding some or all of the following as sub-properties to identifier.  These are industry established ontologies, and while the identifier string is fairly easy to look up, the exact propertyID value may not be as easy to find.  For instance, the example even appears to be case sensitive for propertyID.

**Example using identifier that does not have sub-property.**
~~~~
 "identifier": {
 "@type": "PropertyValue",
   "propertyID": "OCoLC",
   "value":  "889647468"
   },
 ~~~~

Music related:
ISNI: an identifier for persons who publish or contribute to a musical publication https://isni.org/
ISWC:  identifier for writer / publisher of a musical work https://isrc.ifpi.org/en/
ISRC:  identifier for a musical recording http://www.iswc.org/
MUBZ: a music database system sponsored by Google among others.  musicbrainz.org

Additional identifier ontologies and prefixes:
VIAF: Virtual Identifier Authority File   http://viaf.org/
LC: Library of Congress 
WKD: Wikidata
 

Google Knowledge Graph Cards have share links with a dedicated url,  I don't know if they are considered permanent, or if they are meant to be used for disambiguation. 

My reasoning here is why should we have to guess the name side of the name/value pair?  Especially when these are very well-known ontologies.

**Comment by jaygray0919** [comment]
Similar issues in bioinformatics. For example a chemical. There are at least four important identifiers for an item.

1. InChi String

2. InChi Key

3. SMILES

4. CAS (a sometimes ambiguous identifier that needs text explanation to distinguish versions and properties)

While it's not schema.org's responsibility to standardize biochemical information, it would be helpful to support the identification of items like chemicals (or other `@CreativeWork`s)

**Comment by RichardWallis** [comment]
Couple of thoughts on this.

As each domain promotes the benefit of its favoured set of ideal identifier sub-properties, things could rapidly escalate to the point where we are trying manage a significant set of these that is _always one or two missing_.

This is one of the main reasons for having [PropertyValue](https://schema.org/PropertyValue) as an expected type for the _identifier_ property (of _Thing_ and therefore available for all other Types).   This provides a way of describing any identifier(s), possibly using common within an industry acronyms, as per the OcLc example mentioned by @WeaverStever.  Using _PropertyValue_ also makes available several properties such as _description_, _sameAs_, etc. that would be helpful in situations such as the CAS example provided by @jaygray0919.

This is not to say that suggestions for new identifier sub-types should not be made, but they should probably pass scrutiny as being widely known, understood, and used. 


Personally I think ISNI is a potential here as it is used to identify 11 million+ individuals across all areas of creative work publishing from academic papers to music.

**Comment by jeannieh** [comment]
Agreeing with Richard Wallis and Jay.

**Comment by WeaverStever** [comment]
One idea might be to adjust the guidance in propertyID to include a link to the Wikipedia Authority Control template / table. 

**Current propertyID description:**

> A commonly used identifier for the characteristic represented by the property, e.g. a manufacturer or a standard code for a property. propertyID can be (1) a prefixed string, mainly meant to be used with standards for product properties; (2) a site-specific, non-prefixed string (e.g. the primary key of the property or the vendor-specific id of the property), or (3) a URL indicating the type of the property, either pointing to an external vocabulary, or a Web resource that describes the property (e.g. a glossary entry). Standards bodies should promote a standard prefix for the identifiers of properties from their standards.


**Proposed addition:**

> A list of well-known standard prefixes can be found in the table "Authority control identifiers" on this page.
> https://en.wikipedia.org/wiki/Template:Authority_control#Usage



**Note:**
It would be very cool to have a declared sub-property for WikiDataItem (which the above does not cover).  I would use the heck out of it because that's where the data is and should be maintained collectively imho.

**Comment by jaygray0919** [comment]
We agree with the need to avoid the trap of catering to ontology-specific requests. Our thought is to support an array where `@PropertyValue` has a label for the Authority and then the Authority ID. This might be re-enforced using `sameAs` (a three-part structure).

**Comment by WeaverStever** [comment]
@jaygray0919 we were writing at the same time.

I proposed an externally maintained list, see post above.

I agree, propertyIDauthority could be added to the https://schema.org/PropertyValue Type.

**Comment by WeaverStever** [comment]
Continued....

This Category might be a candidate for guidance in the new "propertyIDauthority" item.
https://en.wikipedia.org/wiki/Category:Lists_of_standards

**Comment by alexaguilar89** [comment]
?

**Comment by WeaverStever** [comment]
Revisiting.

Proposal as sub-properties for https://schema.org/identifier.

ISNI:  https://isni.org/  Adopted by YouTube for identifying creators. (In 2018 I believe)

Wikidata: https://www.wikidata.org/wiki/Wikidata:Main_Page

**Comment by davux** [comment]
JSON already provides a native key/value concept, so we should probably use that rather than using hand-made propertyID/value properties in an object.

Moreover, `identifier` like any attribute can be extended. So as an alternative, maybe we could let ISNI/Musicbrainz/etc identifiers be defined as their own attributes that *extend* identifier, rather that giving `identifier` an object structure. What do you think?

**Comment by WeaverStever** [comment]
@davux 

Yes, in music specific types, I found the identifiers are adequately covered in musicComposition and musicRecording.

Conversely, people in these forums have been suggesting a short cut to Wikidata items and I suggest that ISNI is rapidly becoming another candidate for an Identifier sub-property -- mainly because YouTube/Google has adopted ISNI.  I believe that ISNI covers organizations, groups and persons who publish.
